vault/resource_mfa_duo.go

```diff
@@ -0,0 +1,149 @@
+package vault
+
+import (
+	"fmt"
+	"log"
+
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/hashicorp/vault/api"
+)
+
+func mfaDuoResource() *schema.Resource {
+	return &schema.Resource{
+		Create: mfaDuoWrite,
+		Update: mfaDuoWrite,
+		Delete: mfaDuoDelete,
+		Read:   mfaDuoRead,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:         schema.TypeString,
+				Required:     true,
+				Description:  "Name of the MFA method.",
+				ValidateFunc: validateNoTrailingSlash,
+			},
+			"mount_accessor": {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "The mount to tie this method to for use in automatic mappings. The mapping will use the Name field of Aliases associated with this mount as the username in the mapping.",
+			},
+			"username_format": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Description: "A format string for mapping Identity names to MFA method names. Values to substitute should be placed in `{{}}`.",
+			},
+			"secret_key": {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "Secret key for Duo.",
+			},
+			"integration_key": {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "Integration key for Duo.",
```

Comment:
Also, I just realized the secret key and integration key might need to be marked "Sensitive" so they don't pop out in Terraform's output.

Reply:
Done! 👍